runtime/gateway.go

```diff
@@ -305,7 +305,9 @@ func (gateway *Gateway) Bootstrap() error {
 
 	// start TChannel server
 	ip := localhost
-	if env != testenv {
+	if env != testenv && gateway.Config.ContainsKey("tchannel.server.ip") {
```

Comment:
Let's remove the `env != testenv` condition here. It makes more sense for the configuration to use the specific ip address that we provide even when it's in a test environment.